vpr/src/draw/draw_basic.cpp

```diff
@@ -701,6 +749,28 @@ void draw_partial_route(const std::vector<RRNodeId>& rr_nodes_to_draw, ezgl::ren
     }
 }
 
+/* Helper function that checks whether the edges between the current and previous nodes can be drawn
+ * based on whether the cross-layer connections option is enabled and whether the layer on which the
+ * nodes are located are enabled.
+ */
+bool is_edge_valid_to_draw(RRNodeId current_node, RRNodeId prev_node) {
+    t_draw_state* draw_state = get_draw_state_vars();
+    auto& rr_graph = g_vpr_ctx.device().rr_graph;
+
+    int current_node_layer = rr_graph.node_layer(current_node);
+    int prev_node_layer = rr_graph.node_layer(prev_node);
+
+    if (current_node_layer != prev_node_layer) {
+        if (draw_state->cross_layer_display.visible && draw_state->draw_layer_display[current_node_layer].visible && draw_state->draw_layer_display[prev_node_layer].visible) {
+            return true; //if both layers are enabled and cross layer connections are enabled
+        } else {
+            return false; //if cross layer connections are disabled or if either the current or prev node's layers are disabled
+        }
+    } else {
+        return true; //if both nodes are from the same layer
```

Comment:
For safety and ease of commenting, I'd return (draw_state->draw_layer_display[current_node_layer].visible).

(Now it is safe to use even for edges in the same layer.  If you didn't do this, you'd have to comment the function limitation (even though the callers currently don't call it in this case).